server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -224,6 +241,10 @@ public void writeTo(StreamOutput out) throws IOException {
                 out.writeString(timeZone.getId());
                 out.writeVInt(resolution.ordinal());
             }
+            if (out.getVersion().onOrAfter(Version.CURRENT)) {
+                //in order not to loose information if the formatter is a joda we send a flag
+                out.writeBoolean(formatter instanceof JodaDateFormatter);//todo pg consider refactor to isJoda method..
```

Comment:
@polyfractal is this a high perf piece of code? Is a instanceof too expensive here?
this could be refactored to a method on a DateFormatter class

Reply:
Hmm no, I don't think this will ever be called in any kind of high-performance, hot loop.  It will be serialized for each bucket that needs to transfer the formatter, so could be called on the order of thousands to hundred-thousands of times, but not like millions or billions.

I suspect any overhead this introduces will be lost in the noise compared to agg execution themselves.